include/swift/Sema/CSFix.h

```diff
@@ -2027,6 +2030,29 @@ class NotCompileTimeConst final : public ContextualMismatch {
   }
 };
 
+class MustBeCopyable final : public ConstraintFix {
+  Type noncopyableTy;
+
+  MustBeCopyable(ConstraintSystem &cs, Type noncopyableTy, ConstraintLocator *locator);
+
+public:
+  std::string getName() const override { return "remove move-only from type"; }
+
+  bool diagnose(const Solution &solution, bool asNote = false) const override;
+
+  bool diagnoseForAmbiguity(CommonFixesArray commonFixes) const override {
```

Comment:
Should be very careful with this because having the same location doesn’t mean that all fixes point to the same type.

Reply:
After looking at some of the other `ContextualMismatch::diagnoseForAmbiguity`, I think I know what you mean: the types within the fixes across the various solution-fix pairs might be different, so I should verify they're all the same first, otherwise return `false` to indicate that I can't diagnose this ambiguous set of solutions so that at least the "ambiguous solutions" error is emitted.

Reply:
Yeah, some of the fixes are like this, contextual is a good example.

Reply:
Thanks. Just fixed it.